source/Nevermore.IntegrationTests/Legacy/QueryBuilderIntegrationFixture.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Linq;
+using FluentAssertions;
+using Nevermore.IntegrationTests.Legacy.Model;
+using NUnit.Framework;
+
+namespace Nevermore.IntegrationTests.Legacy
+{
+    public class QueryBuilderIntegrationFixture : FixtureWithRelationalStore
+    {
+        [Test]
+        public void WhereInClause()
+        {
+            using (var trn = Store.BeginTransaction())
+            {
+                trn.TableQuery<Product>()
+                    .Where("[Id] IN @ids")
+                    .Parameter("ids", new[] {"A", "B"})
+                    .ToList();
+            }
+        }
+
+        [Test]
+        public void WhereInClauseWhenParameterNamesDifferByCase()
+        {
+            using (var trn = Store.BeginTransaction())
+            {
+                trn.TableQuery<Product>()
+                    .Where("[Id] IN @Ids")
+                    .Parameter("ids", new[] {"A", "B"})
+                    .ToList();
+            }
+        }
+
+        [Test]
+        public void WhereNullClause()
+        {
+            using (var t = Store.BeginTransaction())
+            {
+                foreach (var c in new []
+                {
+                    new Customer {FirstName = "Alice", LastName = "Apple", Nickname = null},
+                    new Customer {FirstName = "Bob", LastName = "Banana", Nickname = ""},
+                    new Customer {FirstName = "Charlie", LastName = "Cherry", Nickname = "Chazza"}
+                })
+                    t.Insert(c);
+                t.Commit();
+                
+                var customersNull = t.TableQuery<Customer>()
+                    .Where(c => c.Nickname == null)
```

Comment:
It might be useful to convert `Nickname` to a `TypedString` to make sure LINQ to Nevermore works with the new library.

Reply:
I've done this 👍 It returns the expected results in the usual case...but with one limitation. Right now, these two filters both work equivalently:
- `.Where(c => c.Nickname == new Nickname("Chazza"))` 
- `.Where(c => c.Nickname == new SomeOtherTypedString("Chazza"))`

Both return documents that have a `Nickname` with value `"Chazza"`. It would be nicer if the latter always returned no results.

This behavior appears to be because command parameter values are unwrapped and eventually happily passed to the `SqlClient.SqlParameterCollection` as plain `strings`.

https://github.com/OctopusDeploy/Nevermore/blob/bd7ab2445ef4f2ac82882e8b04bff5563eb03b0f/source/Nevermore/CommandParameterValues.cs#L90

Reply:
Fixed in https://github.com/OctopusDeploy/Nevermore/pull/87/commits/4589bfeeebcc2a2a851342e180313e05f3cd9741. 

A bit heavy-handed maybe? It affects all queries, not just ones involving tiny types, and would now return no results for any query of the form:
```
Where(c => c.SomeProperty == someValueThatCannotBeAssignedToThatPropertyType)
```

Reply:
Changed my mind, and reverted the above in https://github.com/OctopusDeploy/Nevermore/pull/87/commits/e5c79c6be0fb94ac34acc23aa0dd7741073268c0, so now the SQL filter generated from expressions like `.Where(c => c.Nickname == new SomeOtherTypeOfStringString("Chazza"))` will once again just do a basic SQL comparison on the underlying value.

Reasoning:
1. Both behaviors are potentially surprising (i.e. returning rows matching the underlying value, or returning no rows)
2. Neither behaviors achieve the goal of this work any more than the other, i.e. improving _compile-time_ type safety
3. We can't possibly hope to implement all custom `==` operator implementations in SQL
3. Given that, let's be honest with ourselves and just treat these expressions as syntactic sugar for generating SQL queries which, of course, end up just comparing serialized values